python/tskit/trees.py

```diff
@@ -2209,6 +2209,16 @@ def nodes(self, root=None, order="preorder"):
             for u in roots:
                 yield from iterator(u)
 
+    def _node_edges(self):
+        """
+        Return a numpy array mapping the node IDs in this tree to the ID of the edge
+        above them. This is in lieu of a tree.edge(u) function, currently implemented
+        using the non-optimised Python TreeSequence._tree_node_edges() generator
+        """
+        for index, node_edge_map in enumerate(self.tree_sequence._tree_node_edges()):
```

Comment:
I think we need a ``# pragma: no branch`` here, because this loop never terminates

 https://coverage.readthedocs.io/en/coverage-5.5/branch.html#structurally-partial-branches